pkg/webui/console/views/device-import/index.js

```diff
@@ -12,53 +12,55 @@
 
-import React, { Component } from 'react'
+import React from 'react'
 import { Container, Col, Row } from 'react-grid-system'
-import { connect } from 'react-redux'
+import { useSelector } from 'react-redux'
 import { defineMessages } from 'react-intl'
 
+import { useBreadcrumbs } from '@ttn-lw/components/breadcrumbs/context'
 import Breadcrumb from '@ttn-lw/components/breadcrumbs/breadcrumb'
-import { withBreadcrumb } from '@ttn-lw/components/breadcrumbs/context'
 import Notification from '@ttn-lw/components/notification'
 import PageTitle from '@ttn-lw/components/page-title'
+import ErrorNotification from '@ttn-lw/components/error-notification'
+
+import RequireRequest from '@ttn-lw/lib/components/require-request'
 
 import DeviceImporter from '@console/containers/device-importer'
 
-import PropTypes from '@ttn-lw/lib/prop-types'
 import sharedMessages from '@ttn-lw/lib/shared-messages'
 
+import { listBrands } from '@console/store/actions/device-repository'
+
 import {
   selectDeviceTemplateFormats,
   selectDeviceTemplateFormatsFetching,
 } from '@console/store/selectors/device-template-formats'
+import { selectSelectedApplicationId } from '@console/store/selectors/applications'
 
 const m = defineMessages({
   noTemplatesTitle: 'No end device templates found',
   noTemplates:
     'There are currently no end device templates set up. Please set up an end device template to make use of the bulk device import feature. For more information please refer to the documentation.',
 })
 
-@connect(state => ({
-  deviceTemplateFormats: selectDeviceTemplateFormats(state),
-  deviceTemplateFormatsFetching: selectDeviceTemplateFormatsFetching(state),
-}))
-@withBreadcrumb('devices.import', props => {
-  const { appId } = props
-  return (
-    <Breadcrumb path={`/applications/${appId}/devices/import`} content={sharedMessages.import} />
+const DeviceAddBulk = () => {
+  const appId = useSelector(selectSelectedApplicationId)
+  useBreadcrumbs(
+    'devices.import',
+    <Breadcrumb path={`/applications/${appId}/devices/import`} content={sharedMessages.import} />,
   )
-})
-export default class DeviceAddBulk extends Component {
-  static propTypes = {
-    deviceTemplateFormats: PropTypes.shape().isRequired,
-    deviceTemplateFormatsFetching: PropTypes.bool.isRequired,
-  }
 
-  render() {
-    const { deviceTemplateFormatsFetching, deviceTemplateFormats } = this.props
-    const showEmptyWarning =
-      !deviceTemplateFormatsFetching && Object.keys(deviceTemplateFormats).length === 0
-    return (
+  const deviceTemplateFormats = useSelector(selectDeviceTemplateFormats)
+  const deviceTemplateFormatsFetching = useSelector(selectDeviceTemplateFormatsFetching)
+  const showEmptyWarning =
+    !deviceTemplateFormatsFetching && Object.keys(deviceTemplateFormats).length === 0
+
+  return (
+    <RequireRequest
+      requestAction={listBrands(appId, {}, ['name', 'lora_alliance_vendor_id'])}
+      errorRenderFunction={ErrorNotification}
```

Comment:
```suggestion
      errorRenderFunction={SubViewError}
```
I think this is what should be rendered in case of an error.